db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/BinaryListOperatorNode.java

```diff
@@ -110,9 +110,11 @@ public ValueNode getLeftOperand() {
         return (ValueNode) (singleLeftOperand ? leftOperandList.elementAt(0) : null);
     }
     
-    public boolean allLeftOperandsColumnReferences() {
-        for (Object obj:leftOperandList) {
-            if (!(obj instanceof  ColumnReference))
+    public boolean allLeftOperandsContainColumnReferences() throws StandardException {
+        CollectNodesVisitor cnv = new CollectNodesVisitor(ColumnReference.class);
+        for (int i = 0; i < leftOperandList.size(); i++) {
+            leftOperandList.elementAt(i).accept(cnv);
+            if (cnv.getList().isEmpty())
```

Comment:
The variable declaration of `cnv` should be moved inside the for loop, otherwise, we won't be able to detect empty column references correctly for second element and upper.

Reply:
Ah, yes, this is buggy. I add `cnv.getList().clear()` in the loop. It should clear the vector and avoid constructing new object for each iteration.